cmd/minikube/cmd/cp.go

```diff
@@ -83,6 +87,35 @@ Example Command : "minikube cp a.txt /home/docker/b.txt" +
 func init() {
 }
 
+// destinationPath sets the src filename as dst filename
+// when the dst file name is not provided and ends with a `/`.
+// Otherwise this function is a no-op and returns the passed dst.
+func destinationPath(dst, src string) string {
+	dd, df := filepath.Split(dst)
+
+	// if dst is empty, dd and df will be empty, so return dst
+	// validation will be happening in `validateArgs`
+	if dd == "" && df == "" {
+		return ""
+	}
+
+	// if filename is already provided, return dst
+	if df != "" {
+		return dst
+	}
+
+	_, sf := filepath.Split(src)
+
+	// if src filename is empty, return dst
+	if sf == "" {
+		return dst
+	}
+
+	// if [dst] filename is not provided and [src] filename is provided,
+	// join [dst] dir and [src] filename
+	return filepath.Join(dd, sf)
```

Comment:
This is going to fail on Windows.

Example:
```
# on Windows
$ minikube cp /var/lib/minikube/ /var/cat123

# will result in
s := destinationPath("/var/lib/minikube/", "/var/cat123")
fmt.Println(s)
\var\lib\minikube\cat123
```

It's going to use Windows `\` due to `filepath.Join` which uses the OS specific separator. There will need to be some detection whether it's copying to minikube or the host. If it's minikube it should use `file.Join` and if it's the host it should use `filepath.Join`.

Reply:
@spowelljr From the description mentioned in the `cp` command, It seems that file can only be copied into Minikube and not to host. So destination would always be linux right? Hence `filepath.Join` should be ok right?

Reply:
1. The `cp` description is incorrect, you can copy from the machine to the host.
```
minikube cp minikube:<file> <host_dir>
```

2. When on Windows, copying from the host to the machine will fail as it's going to set the destination using `filepath.Join` which uses the OS file separator, which in Windows is `\`, but we're copying into minikube, which needs `/`.

ie.
```
$ minikube cp C:\Users\user\file.txt /var/lib/minikube

filepath.Join(dd, sf) = \var\lib\minikube\file.txt
```

If only copying from the host to minikube could use `path.Join` instead as that would always use `\`, but to allow copying both ways need to use `path.Join` when copying to minikube, otherwise use `filepath.Join`.

Reply:
Understood. Thanks. I will make the change

Reply:
For copying from minikube to host, does the user have to specify the minikube node always in `src`? 

```bash
$ minikube cp <file> <host_dir> # => This won't infer controlplane node as default right?
```

Reply:
Yes, by default it assumes host -> guest

Reply:
Pushed the change